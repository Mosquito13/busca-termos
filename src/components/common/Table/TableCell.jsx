import PropTypes from 'prop-types';
import classNames from 'classnames';

import CopyTextButton from '../CopyTextButton';

const SMALL = 'small';
const AUTO = 'auto';

const TableCell = ({ size, value, header, onSelect, selected, id, itemId, hasCopyButton, odd }) => {
  const classes = classNames(
    'table__cell',
    odd && 'table__cell--odd',
    header && 'table__cell--header',
    selected && 'table__cell--selected',
    size === SMALL && 'table__cell--small'
  );

  return (
    <div className={classes} onClick={() => onSelect?.(itemId, id, value)}>
      <span className="table__cell-text">{value}</span>
      {hasCopyButton && !header && (
        <div className="table__cell-copy-button">
          <CopyTextButton
            value={value}
            noMarginRight={size === SMALL}
          />
        </div>
      )}
    </div>
  );
};

TableCell.propTypes = {
  /**
   * Valor a ser exibido
   */
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  /**
   * Tamanho da celula
   */
  size: PropTypes.oneOf([SMALL, AUTO]),
  /**
   * Indica se faz parte do cabecalho da tabela
   */
  header: PropTypes.bool,
  /**
   * Funcao executada ao selecionar uma celula
   */
  onSelect: PropTypes.func,
  /**
   * Indica se esta celula selecionada
   */
  selected: PropTypes.bool,
  /**
   * Identificador unico da celula
   */
  id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  /**
   * Identificador unico do registro (linha)
   */
  itemId: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  /**
   * Indica se e linha impar (para usar a zebra)
   */
  odd: PropTypes.bool,
  /**
   * Indica se exibe botao de copia na celula
   */
  hasCopyButton: PropTypes.bool
};

export { SMALL, AUTO };
export default TableCell;
