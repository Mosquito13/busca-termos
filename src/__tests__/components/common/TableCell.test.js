import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import TableCell, { SMALL } from '../../../components/common/Table/TableCell';

describe(`${TableCell.name}`, () => {
  describe('tests', () => {
    let wrapper, onSelectSpy;

    beforeAll(() => {
      onSelectSpy = jest.fn();

      wrapper = shallow(
        <TableCell
          value="A value"
          onSelect={onSelectSpy}
          id="666_123"
          itemId="666"
        />
      );
    });

    it('should render', () => {
      expect(wrapper.isEmptyRender()).toBeFalsy();
      expect(wrapper.find('.table__cell')).toHaveLength(1);
      expect(wrapper.text()).toContain('A value');
    });

    it('should fire onSelect', () => {
      wrapper.find('.table__cell').simulate('click');
      expect(onSelectSpy).toBeCalledTimes(1);
      expect(onSelectSpy).toBeCalledWith('666', '666_123', 'A value');
    });

    afterAll(() => {
      wrapper.unmount();
    });
  });

  describe('snapshots', () => {
    it('should render', () => {
      const tree = renderer.create(<TableCell value="123" />).toJSON();

      expect(tree).toMatchSnapshot();
    });

    it('should render with odd modifier', () => {
      const tree = renderer.create(<TableCell odd value="123" />).toJSON();

      expect(tree).toMatchSnapshot();
    });

    it('should render with header modifier', () => {
      const tree = renderer.create(<TableCell header value="123" />).toJSON();

      expect(tree).toMatchSnapshot();
    });

    it('should render with selected modifier', () => {
      const tree = renderer.create(<TableCell selected value="123" />).toJSON();

      expect(tree).toMatchSnapshot();
    });

    it('should render with small modifier', () => {
      const tree = renderer
        .create(<TableCell size={SMALL} value="123" />)
        .toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
