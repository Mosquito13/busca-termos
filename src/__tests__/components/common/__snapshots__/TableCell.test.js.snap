// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableCell snapshots should render 1`] = `
<div
  className="table__cell"
  onClick={[Function]}
>
  <span
    className="table__cell-text"
  >
    123
  </span>
</div>
`;

exports[`TableCell snapshots should render with header modifier 1`] = `
<div
  className="table__cell table__cell--header"
  onClick={[Function]}
>
  <span
    className="table__cell-text"
  >
    123
  </span>
</div>
`;

exports[`TableCell snapshots should render with odd modifier 1`] = `
<div
  className="table__cell table__cell--odd"
  onClick={[Function]}
>
  <span
    className="table__cell-text"
  >
    123
  </span>
</div>
`;

exports[`TableCell snapshots should render with selected modifier 1`] = `
<div
  className="table__cell table__cell--selected"
  onClick={[Function]}
>
  <span
    className="table__cell-text"
  >
    123
  </span>
</div>
`;

exports[`TableCell snapshots should render with small modifier 1`] = `
<div
  className="table__cell table__cell--small"
  onClick={[Function]}
>
  <span
    className="table__cell-text"
  >
    123
  </span>
</div>
`;
